submit_cluster_job.py

```diff
@@ -0,0 +1,128 @@
+"""
+Submit a cluster LSF job to train a neural network.
+
+e.g.:
+    python submit_cluster_job.py --pipeline train.py --configuration configuration.yaml
+
+    python submit_cluster_job.py --gpu V100 --pipeline train.py --configuration configuration.yaml
+"""
+
+
+# standard library imports
+import argparse
+import datetime as dt
+import pathlib
+import shutil
+import subprocess
+import sys
+
+# third party imports
+import yaml
+
+from pytorch_lightning.utilities import AttributeDict
+
+
+def main():
+    """
+    main function
+    """
+    argument_parser = argparse.ArgumentParser()
+    argument_parser.add_argument("--pipeline", type=str, help="pipeline script path")
+    argument_parser.add_argument(
+        "--configuration", type=str, help="experiment configuration file path"
+    )
+    argument_parser.add_argument(
+        "--gpu",
+        default="A100",
+        choices=["A100", "V100"],
+        type=str,
+        help="GPU nodes queue to submit job",
+    )
+    argument_parser.add_argument(
+        "--num_gpus", default=1, type=int, help="number of GPUs to use"
+    )
+    argument_parser.add_argument(
+        "--mem_limit", default=65536, type=int, help="GPU node RAM memory limit"
+    )
+
+    args = argument_parser.parse_args()
+
+    # submit new training job
+    if args.pipeline and args.configuration:
+        datetime = dt.datetime.now().isoformat(sep="_", timespec="seconds")
+
+        pipeline_path = pathlib.Path(args.pipeline)
+
+        with open(args.configuration) as file:
```

Comment:
should `args.configuration` be moved to `train.py`? As configuration contain all of the hyperparameters and `train.py` need those.